pkg/controllers/operator/cilium-crds/endpoint/endpoint_controller.go

```diff
@@ -533,7 +533,7 @@ func (r *endpointReconciler) handlePodUpsert(ctx context.Context, newPEP *PodEnd
 		// May end up getting another endpoint ID below if we try to create the CEP below.
 		// No downside to this.
 
-		if !k8serrors.IsNotFound(err) && err != nil {
```

Comment:
this change is unrelated but was quick so did it, we have already checked this condition before so it was an always true condition